hugr-core/src/extension/op_def.rs

```diff
@@ -115,13 +115,12 @@ pub trait CustomLowerFunc: Send + Sync {
     ) -> Option<Hugr>;
 }
 
-/// Encode a signature as `PolyFuncTypeRV` but optionally allow validating type
+/// Encode a signature as `PolyFuncTypeRV` but allow validating type
```

Comment:
```suggestion
/// Encode a signature as [PolyFuncTypeRV] but with additional (custom) validation of type args
```

We don't just allow validating, we require it!